src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,53 @@
 package core.basesyntax.service;
 
+import core.basesyntax.RegistrationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LENGTH = 6;
+    private static final int MINIMAL_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        return storageDao.add(checkUserValidation(user));
+    }
+
+    public User checkUserValidation(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("Invalid data");
+        }
```

Comment:
Consider adding a more descriptive error message for when the user object is null, such as 'User object cannot be null'.